### PR TITLE
add doc for codegen of CRD

### PR DESCRIPTION
**Pull request by m3ngyang**
fix issue #582

### CODE REVIEW

File: doc/edl/README.md [hunk 1]
```diff
@@ -159,6 +159,26 @@ implementing custom resources:
 We will support TPR first, because some of our clusters is using
 Kubernetes v1.6. CRD will be supported in the future.
 
+In the near feature, we will use CRD to replace TPR to define our training job resource,
+then the resource defined under `go/edl/resource` will be deprecated.
+If you want to use TPR in Kubernetes with low version,
+please checkout project to the tag `unreleased-tpr`.
```
Comment (typhoonzero) [inline doc/edl/README.md]: please checkout the `unreleased-tpr` tag.

File: doc/edl/README.md [hunk 2]
```diff
@@ -159,6 +159,26 @@ implementing custom resources:
 We will support TPR first, because some of our clusters is using
 Kubernetes v1.6. CRD will be supported in the future.
 
+In the near feature, we will use CRD to replace TPR to define our training job resource,
+then the resource defined under `go/edl/resource` will be deprecated.
+If you want to use TPR in Kubernetes with low version,
```
Comment (typhoonzero) [inline doc/edl/README.md]: with low version => with version <= 1.7

File: doc/edl/README.md [hunk 3]
```diff
@@ -159,6 +159,26 @@ implementing custom resources:
 We will support TPR first, because some of our clusters is using
 Kubernetes v1.6. CRD will be supported in the future.
```
Comment (typhoonzero) [inline doc/edl/README.md]: "CRD will be supported in the future." can be removed now.

File: doc/edl/README.md [hunk 4]
```diff
@@ -159,6 +159,26 @@ implementing custom resources:
...
 
+In the near feature, we will use CRD to replace TPR to define our training job resource,
+then the resource defined under `go/edl/resource` will be deprecated.
+If you want to use TPR in Kubernetes with low version,
+please checkout project to the tag `unreleased-tpr`.
+
+Currently, implementation with CRD is still under development,
```
Comment (typhoonzero) [inline doc/edl/README.md]: implementation with CRD => implementation using CRD

File: doc/edl/README.md [hunk 5]
```diff
@@ -159,6 +159,26 @@ implementing custom resources:
...
+you cloud use the following command to check and generate dependent codes:
```
Comment (typhoonzero) [inline doc/edl/README.md]: cloud => can

to check and generate dependent codes: => to verify and ...

File: doc/edl/README.md [hunk 6]
```diff
@@ -159,6 +159,26 @@ implementing custom resources:
...
+```
+# check freshness of generated codes
+$ scripts/verify-codegen.sh
+
+# update dependent codes
+$ scripts/update-codegen.sh
+```
+For more detailed operation, please refer to [article](https://blog.openshift.com/kubernetes-deep-dive-code-generation-customresources/).
```
Comment (typhoonzero) [inline doc/edl/README.md]: For more detailed operation => for more details

File: doc/edl/README.md [hunk 7]
```diff
@@ -159,6 +159,26 @@ implementing custom resources:
...
+When the definition of resource is stable, we will commit generated dependenct codes.
```
Comment (typhoonzero) [inline doc/edl/README.md]: typo "dependenct"

File: doc/edl/README.md [hunk 8]
```diff
@@ -159,6 +159,26 @@ implementing custom resources:
...
+Just skip `go/pkg` when running `go test` during your development.
```
Comment (typhoonzero) [inline doc/edl/README.md]: I think we are removing `go/pkg` directory now?

Comment (m3ngyang) [reply doc/edl/README.md]: yes, but we still have this currently. When I fix conflicts in #587 I will change the package name here.

**Pull request merged**